src/thermo/Phase.cpp

```diff
@@ -559,6 +559,13 @@ const vector_fp& Phase::molecularWeights() const
     return m_molwts;
 }
 
+const vector_fp& Phase::getSpeciesCharges(double* charges) const
+{
+    const vector_fp& ch = m_speciesCharge;
+    copy(ch.begin(), ch.end(), charges);
```

Comment:
There's no need to make a reference to `m_speciesCharge`. You can just write
```suggestion
    copy(m_speciesCharge.begin(), m_speciesCharge.end(), charges);
```